### PR TITLE
HDFans:add new Private Tracker resolves #13272

### LINKED ISSUE #13272: [req]: HDFans
### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Is there already a request for your tracker?

- [X] I have checked older issues, open and closed

### Type

Private

### Tracker details

```markdown
- Name: HDFans
- URL: https://hdfans.org/
- Description: Chinese general tracker
- Minimum Seeding Time and Minimum Ratio (if Private): 
- Additional Comments:
```
Registration currently closed, invites only
engine: NexusPHP 1.7.6

### PR DESCRIPTION
**Pull request by sandboxdream**
resolves #13272

### CODE REVIEW

File: src/Jackett.Common/Definitions/hdfans.yml [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    - {id: 405, cat: TV/Anime, desc: "Animations/动漫"}
+    - {id: 402, cat: TV, desc: "TV Series/剧集"}
+    - {id: 403, cat: TV, desc: "TV Shows/综艺"}
+    - {id: 404, cat: Other, desc: "Education/教育"}
+    - {id: 406, cat: Audio/Video, desc: "MusicVideo/音乐MV"}
+    - {id: 407, cat: TV/Sport, desc: "Sports/运体"}
+    - {id: 409, cat: Other, desc: "Misc/其他"}
+    - {id: 408, cat: Audio, desc: "HQ Audio/无损音乐"}
+    - {id: 419, cat: Other, desc: "Software/软件"}
+    - {id: 421, cat: Other, desc: "Games/游戏"}
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdfans.yml:24]: Please find the most appropriate categories - https://github.com/Jackett/Jackett/wiki/Jackett-Categories
Suggestion below:
```suggestion
    - {id: 419, cat: PC/0day, desc: "Software/软件"}
    - {id: 421, cat: PC/Games, desc: "Games/游戏"}
```

File: src/Jackett.Common/Definitions/hdfans.yml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+  modes:
+    search: [q]
+    tv-search: [q, season, ep, imdbid, doubanid]
+    movie-search: [q, imdbid, doubanid]
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdfans.yml:29]: Is doubanid searching supported? If so, you'd need to implement it in the search block, e.g. - https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Definitions/52pt.yml

**Review by ilike2burnthing — CHANGES_REQUESTED** [review]
Thanks for your contribution, please see the requested changes.

File: src/Jackett.Common/Definitions/hdfans.yml (hunk 1 above)
Comment (garfield69) [reply src/Jackett.Common/Definitions/hdfans.yml:24]: Please review your categories as they are not a perfect match for what is on the site.
I found the following in the **User CP > Tracker Settings** page:
401 Movies/电影
402 TV Series/电视剧"
403 Documentaries/纪录片
406 Music/音乐
416 TV Shows/综艺
417 Animations/动漫、动画
407 Music Videos/音乐视频
408 Concert/演唱会
404 Education/教育
405 Audio Books/有声读物
409 Drama/戏剧
418 Sports/体育
419 Software/软件
421 Games/游戏
423 E-Books/电子书
410 Others/其他

**Review by garfield69 — CHANGES_REQUESTED** [review]

File: src/Jackett.Common/Definitions/hdfans.yml [hunk 3]
```diff
@@ -0,0 +1,179 @@
+---
+id: hdfans
+name: HDFans
+description: "HDFans is a CHINESE Private Torrent Tracker for HD MOVIES / TV"
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdfans.yml:4]:
```suggestion
description: "HDFans is a CHINESE Private Torrent Tracker for MOVIES / TV / GENERAL"
```

File: src/Jackett.Common/Definitions/hdfans.yml [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    movie-search: [q, imdbid, doubanid]
+    music-search: [q]
+
+settings:
+  - name: username
+    type: text
+    label: Username
+  - name: password
+    type: password
+    label: Password
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdfans.yml:38]: Also add 2FA, e.g. https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Definitions/carpt.yml

File: src/Jackett.Common/Definitions/hdfans.yml [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+  path: login.php
+  method: form
+  form: form[action="takelogin.php"]
+  captcha:
+    type: image
+    selector: img[alt="CAPTCHA"]
+    input: imagestring
+  inputs:
+    username: "{{ .Config.username }}"
+    password: "{{ .Config.password }}"
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdfans.yml:74]: Also add 2FA

File: src/Jackett.Common/Definitions/hdfans.yml [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+      case:
+        img.pro_50pctdown2up: 2
+        img.pro_free2up: 2
+        img.pro_2up: 2
+        "*": 1
+    description:
+      selector: td:nth-child(2)
+      remove: a, img
+    minimumratio:
+      text: 0.81
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdfans.yml:178]:
```suggestion
```
I didn't see any mention of an MR in the rules or FAQ.

**Review by ilike2burnthing — CHANGES_REQUESTED** [review]
Just a few extra changes.

**Comment by garfield69** [comment]
darn, blinking github was showing duplicate posts, and when I deleted one of them it blew both of them away!

**Review by garfield69 — CHANGES_REQUESTED** [review]
401 Movies/电影
402 TV Series/电视剧
403 Documentaries/纪录片
406 Music/音乐
416 TV Shows/综艺
417 Animations/动漫、动画
407 Music Videos/音乐视频
408 Concert/演唱会
404 Education/教育
405 Audio Books/有声读物
409 Drama/戏剧
418 Sports/体育
419 Software/软件
421 Games/游戏
423 E-Books/电子书
410 Others/其他

File: src/Jackett.Common/Definitions/hdfans.yml (hunk 5 above)
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdfans.yml:74]: Also add 2FA here

File: src/Jackett.Common/Definitions/hdfans.yml (hunk 2 above)
Comment (sandboxdream) [reply src/Jackett.Common/Definitions/hdfans.yml:29]: it isn't support douban id .it only support imdbid to search

Comment (ilike2burnthing) [reply src/Jackett.Common/Definitions/hdfans.yml:29]: Ok, then you can just remove doubanid searching.

File: src/Jackett.Common/Definitions/hdfans.yml (hunk 6 above)
Comment (sandboxdream) [reply src/Jackett.Common/Definitions/hdfans.yml:178]: it mentioned on faq(https://hdfans.org/faq.php)
>Peasant  被降级的用户，他们有30天时间来提升分享率，否则他们会被踢。不能发表趣味盒内容；不能申请友情链接；不能上传字幕。
>当以下情况时将被自动降至本级：1.如果你已经下载了超过50GB，你应该有大于0.4的分享率。2.如果你已经下载了超过100GB，你应该有大于0.5的分享率。3.如果你已经下载了超过200GB，你应该有大于0.6的分享率。4.如果你已经下载了超过400GB，你应该有大于0.7的分享率。5.如果你已经下载了超过800GB，你应该有大于0.8的分享率。

it means that if you are download over 50GB you need 0.4Ratio. If you are download over 100GB your ratio need over 0.6............. if you are download over 800GB your ratio need over 0.8.

If you can't exchange ratio over that for 30days you will be banned

Comment (ilike2burnthing) [reply src/Jackett.Common/Definitions/hdfans.yml:185]: Weirdly that section doesn't exist when using the site in English.

File: src/Jackett.Common/Definitions/hdfans.yml [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+    - {id: 407, cat: TV/Sport, desc: "Sports/运体"}
+    - {id: 408, cat: Audio, desc: "HQ Audio/无损音乐"}
+    - {id: 419, cat: PC/0day, desc: "Software/软件"}
+    - {id: 421, cat: PC/Games, desc: "Games/游戏"}
+
+  modes:
+    search: [q]
+    tv-search: [q, season, ep, imdbid]
+    movie-search: [q, imdbid]
+    music-search: [q]
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdfans.yml:29]: Once you've added the additional categories as highlighted by garfield69, you need to add the book-search mode:
```suggestion
    music-search: [q]
    book-search: [q]
```

**Comment by ilike2burnthing** [comment]
Extra categories - https://github.com/Jackett/Jackett/pull/13452#pullrequestreview-1064351710

Category titles - https://github.com/Jackett/Jackett/wiki/Jackett-Categories

**Comment by sandboxdream** [comment]
I don't know the category should use id on website or id on  [wiki](https://github.com/Jackett/Jackett/wiki/Jackett-Categories)

**Comment by ilike2burnthing** [comment]
> I don't know the category should use id on website or id on [wiki](https://github.com/Jackett/Jackett/wiki/Jackett-Categories)

Website.

**Review by ilike2burnthing — APPROVED** [review]

**Pull request merged**